### ISSUE TITLE
Pass APPS key through to the application

### ISSUE THREAD

**Issue by PhilipDaniels** [mintty/mintty#494]
Please, could we have an option to pass the APPS key (typically found to the right of the right Windows key, with a picture of a menu on it) through to the underlying application rather than popping up the menu?

I would like to bind a key to the "super" prefix in Emacs and this is the best candidate, but it is currently intercepted by mintty.

**Comment by mintty** [comment]
Can you test the uploaded version (from Download ZIP button) please?

**Comment by mintty** [comment]
This reveals a bug in my implementation; the menu key should of course *not* be suppressed by default (fix uploaded).
It should rather be configured in your .minttyrc, as described in the manual page.
Example, to generate the same sequence as xterm uses by default:
Key_Menu=29

**Comment by PhilipDaniels** [comment]
This is better, thanks. Terminal Emacs receives the key and interprets it as "<print>" and it can be bound as a prefix key using something like

```
(global-set-key (kbd "<print> r") 'recentf-open-files)
```

I don't know there is a way to get Cygwin's terminal Emacs to see <print> as the w2-apps-modifier so that the piece of setq code above works. I am not sure how Emacs maps its keyboard input to its internal keysyms or whether terminfo etc is involved in the translation, so I can't say whether this is *correct*, but it is certainly usable. Let me do some more digging tonight to see if I can figure out exacly what is happening. I may need to ask a question on Emacs stack exchange.

**Comment by PhilipDaniels** [comment]
Question on stack overflow: http://emacs.stackexchange.com/questions/18245/making-terminal-emacs-treat-apps-key-as-super-modifier

**Comment by PhilipDaniels** [comment]
Hi, I think you can close this issue. The code I worked out seems to be the best that is possible in Emacs. Pasting below for completeness (assuming Key_Menu = 29)

Thanks for your efforts,

```
(if (equal system-type 'cygwin)
     (if (equal window-system 'w32)
         (setq w32-pass-apps-to-system nil
               w32-apps-modifier nil)
       (progn ;; force all alternatives to <apps> so we can write one set of keybindings.
         (define-key key-translation-map (kbd "<print>") (kbd "<apps>"))
         (define-key key-translation-map (kbd "<menu>") (kbd "<apps>")))))

 (define-key global-map (kbd "<apps> h")
    (lambda () (interactive) (message "hello from menu key via <apps> leader key")))
```

**Comment by mintty** [comment]
Released in 2.2.3.
